rclcpp/topics/minimal_subscriber/member_function_with_type_adapter.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "rclcpp/type_adapter.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+using std::placeholders::_1;
+
+template<>
+struct rclcpp::TypeAdapter<std::string, std_msgs::msg::String>
```

Comment:
Can we have a comment here, and in the publisher equivalent, that normally a specialization like this would be in a header and re-used by the publisher and subscription rather than copy-pasted like this? I still think it's good as-is because it keeps us from sharing files between the examples and makes them more "self-contained".